detector/detection_results.go

```diff
@@ -36,6 +37,15 @@ func (r *DetectionResults) Fail(filePath git_repo.FilePath, message string) {
 	}
 }
 
+func (r * DetectionResults) Warn(filePath git_repo.FilePath, message string) {
+	warnings, ok := r.warnings[filePath]
+	if !ok {
+		r.warnings[filePath] = []string{message}
+	} else {
+		r.warnings[filePath] = append(warnings, message)
+	}
+}
+
```

Comment:
There are some merge conflicts in this PR. Please address them.

Reply:
Rectified.